cs251tk/student/markdownify/check_submit_date.py

```diff
@@ -0,0 +1,31 @@
+import os
+from dateutil.parser import parse
+from ...common import run, chdir
+
+
+def check_dates(spec_id, username, spec, basedir):
+    """ Port of the CheckDates program from C++
+        Finds the first submission date for an assignment
+        by comparing first commits for all files in the spec
+        and returning the earliest """
+
+    basedir = os.path.join(basedir, 'students', username, spec_id)
+    dates = []
+
+    with chdir(basedir):
+        for file in spec['files']:
+
+            # Run a git log on each file with earliest commits listed first
+            _, res, _ = run(['git', 'log', '--reverse',

```

Comment:
Should we use `git log --reverse --porcelain` here? My understanding is that the `--porcelain` flag's output is more stable across git releases, and that it's intended for parsing by scripts.

Reply:
It seems that --porcelain isn't a flag for git log. There is a --word-diff=porcelain tag, but that doesn't seem to change anything unless I include the -p/--patch flag.

Reply:
Ah, ok. Thanks for investigating.

Reply:
Next question: Does `git log --reverse --pretty=format:"%ad" --date=iso8601` get you all the information you need? That's just the date, formatted in ISO8601 format, for each commit.

Reply:
I think it does. Let me test it.